finality-provider/service/fp_instance.go

```diff
@@ -194,52 +193,51 @@ func (fp *FinalityProviderInstance) IsRunning() bool {
 func (fp *FinalityProviderInstance) finalitySigSubmissionLoop() {
 	defer fp.wg.Done()
 
+	var targetHeight uint64
 	for {
 		select {
 		case b := <-fp.poller.GetBlockInfoChan():
```

Comment:
I think we can put it in the default case where we have `getAllBlocksFromChan()` to drain the channel and then try to process each block and do batch send

Reply:
Do you mean to put the `var targetHeight uint64` in the `default` case of the inner loop?

Reply:
I meant the entire block fetching logic can be put in the `default` case so that we can have a method to fetch them all other than `fetch one, and the rest`

Reply:
Got it